src/main/java/org/folio/service/consortium/ConsortiumConfigurationService.java

```diff
@@ -74,4 +81,28 @@ public Future<RequestContext> cloneRequestContextIfNeeded(RequestContext request
       .map(config -> config.isEmpty() ? requestContext : RequestContextUtil.createContextWithNewTenantId(requestContext, location.getTenantId()));
   }
 
+  public Future<RequestContext> overrideContextToCentralTenantIdNeeded(RequestContext requestContext) {
+    return getConsortiumConfiguration(requestContext)
+      .compose(consortiumConfiguration -> {
+        if (consortiumConfiguration.isEmpty()) {
+          return Future.succeededFuture(requestContext);
+        }
+        String tenantId = TenantTool.tenantId(requestContext.getHeaders());
+        var configuration = consortiumConfiguration.get();
+        logger.info("overrideContextToCentralTenantIdNeeded:: tenantId from request: {}, centralTenantId: {}",
+          tenantId, configuration.centralTenantId());
+        if (StringUtils.equals(tenantId, configuration.centralTenantId())) {
+          return Future.succeededFuture(requestContext);
+        }
+        RequestContext centralContext = createContextWithNewTenantId(requestContext, configuration.centralTenantId());
+        return settingsRetriever.getSettingByKey(SettingKey.CENTRAL_ORDERING_ENABLED, centralContext)
+          .map(centralOrdering -> {
+            logger.info("overrideContextToCentralTenantIdNeeded:: central ordering enabled: {}", centralOrdering);
+            return centralOrdering.map(Setting::getValue).orElse(null);
+          })
+          .compose(orderingEnabled -> Boolean.parseBoolean(orderingEnabled) ?
+            Future.succeededFuture(centralContext) :
+            Future.succeededFuture(requestContext));
```

Comment:
Can be simplified if you think this is more readable

```suggestion
          .compose(orderingEnabled -> Future.succeededFuture(Boolean.parseBoolean(orderingEnabled) ? centralContext : requestContext);
```

Reply:
thanks, applied recomendation